lib/eslint/flat-eslint.js

```diff
@@ -731,7 +731,21 @@ class FlatESLint {
      * @returns {Promise<LintResult[]>} The results of linting the file patterns given.
      */
     async lintFiles(patterns) {
-        if (!isNonEmptyString(patterns) && !isArrayOfNonEmptyString(patterns)) {
+
+        let normalizedPatterns = patterns;
+
+        /*
+         * Special cases:
+         * 1. `patterns` is an empty string
+         * 2. `patterns` is an empty array
+         *
+         * In both cases, we use the cwd as the directory to lint.
+         */
+        if (patterns === "" || Array.isArray(patterns) && patterns.length === 0) {
+            normalizedPatterns = ["."];
+        } else if (typeof patterns === "string") {
+            normalizedPatterns = [patterns];
+        } else if (!isArrayOfNonEmptyString(patterns)) {
             throw new Error("'patterns' must be a non-empty string or an array of non-empty strings");
         }
```

Comment:
Maybe something like this so that strings with only spaces like `" "` are caught by the validation as it used to be before this change:

```suggestion
        } else {
            if (!isNonEmptyString(patterns) && !isArrayOfNonEmptyString(patterns)) {
                throw new Error("'patterns' must be a non-empty string or an array of non-empty strings");
            }

            if (typeof patterns === "string") {
                normalizedPatterns = [patterns];
            }
        }
```